blog/sourcify-v2/index.mdx

```diff
@@ -0,0 +1,142 @@
+
+```ts
+const pathBuffers: PathBuffer[] = [];
+pathBuffers.push({
+  path: filePath,
+  buffer: fs.readFileSync(filePath),
+});
+const checkedContracts: CheckedContract[] = await checkFiles(pathBuffers);
+```
+
+Then you can verify this `CheckedContract` against a contract that is deployed on a **chain** at an **address**.
+
+```ts
+const goerliChain =   {
+  name: "Goerli",
+  rpc: [
+    "https://locahlhost:8545/"
+    "https://goerli.infura.io/v3/${INFURA_API_KEY}",
+  ],
+  chainId: 5,
+},
+
+const match = await verifyDeployed(
+  checkedContract[0],
+  goerliChain,
+  '0x00878Ac0D6B8d981ae72BA7cDC967eA0Fae69df4'
+)
+
+console.log(match.status) // 'perfect'
+```
+
+### Context Variables
+
+In [previous versions](https://github.com/ethereum/sourcify/releases/tag/v1.2.0) we introduced ["verification with simulation"](https://github.com/ethereum/sourcify/pull/864), that is, executing the contract's compiled creation bytecode on an EVM to see if it will yield the same **onchain deployed bytecode** at the given chainId & address. This allows us to verify "contracts with immutables created by factory contracts".
+
+With this we let the user pass `contextVariables` to the "simulation". These variables are useful when they are used as a value to an immutable variable in the contract's constructor. Currently you can pass two types of variables:
+
+```ts
+const contextVariables = {
+  msgSender: '0x00878Ac0D6B8d981ae72BA7cDC967eA0Fae69df4',
+  abiEncodedConstructorArguments: "0x0000000000000000000000000000000000000000000000000000000000038efe",
+}
+
+const match = await verifyDeployed(
+  checkedContract[0],
+  goerliChain,
+  '0x00878Ac0D6B8d981ae72BA7cDC967eA0Fae69df4'
+  contextVariables
+)
+```
+
+(_In the server API, find the field `contextVariables`_)
+
+For example, if you have a contract like this:
+
+```solidity
+contract Child2{
+    address immutable public owner;
+    constructor(){
+       owner = msg.sender;
+    }
+}
+```
+
+passing the `msgSender` variable allows `lib-sourcify` to assign this variable inside the bytecode, and be able to verify. If there are other potential variables that are used as a value to immutables in contracts, please let us know.
+
+### Creator Tx
+
+We can also verify contracts by looking at the `tx.input` of the transaction that created the contract. If this matches the creation bytecode of the compiled contract AND the address resulting from the `tx.from` and `tx.nonce` matches the given address, we can verify the contract.
+
+```ts
+const match = await verifyDeployed(
+  checkedContract[0],
+  goerliChain,
+  "0x00878Ac0D6B8d981ae72BA7cDC967eA0Fae69df4".undefined,
+  "0xe75fb554e433e03763a1560646ee22dcb74e5274b34c5ad644e7c0f619a7e1d0" //tx hash
+);
+```
+
+(_In the server API, find the field `creatorTx`_)
+
+### CREATE2
+
+You can also verify contracts that are deployed using the CREATE2 opcode:
```

Comment:
this is not accurate, we can also verify contracts that are not yet deployed.

Reply:
Yes good catch